### PR TITLE
[Debt] Split off user auth type

### LINKED ISSUE #8288: ♻️ Get User Auth Info from its own API query
## ♻️ Debt/Refactor

The user model is too big - it encompasses multiple use cases. In particular, updateUser mutations do too much - it makes permission checks complicated. But also, it makes caching efficiently difficult. For example, updating a detail of your profile may lead to refreshing of authentication details, which is unnecessary.

## 🕵️ Details

- Create a new type which holds only and all info the client needs for authentication and authorization (ie sub and roles)
- Add the new type to User model, but also create a new myAuth query, which returns only the auth info for current logged in user
- Use the new query on the frontend

## 🙋‍♀️ Proposed Solution

```
type UserAuthInfo {
  id: UUID!
  sub: String
  roleAssignments: [RoleAssignment]

  createdAt: DateTime
  deletedAt: DateTime
}

type User {
  id: UUID!
  #sub - remove this
  #roleAssignments - remove this
  authInfo: UserAuthInfo
  ... #other existing fields
}

type Query {
  myAuth: UserAuthInfo @auth
  ...
}
```

## ✅ Acceptance Criteria

- [ ] Frontend authentication/authorization gets sub and roles from a new myAuth query which returns only auth data
- [ ] authProvider component should use this myAuth query instead of me query
- [ ] User table should be updated to get roleAssignments from the nested UserAuthInfo instead of User object

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #8288

## 👋 Introduction

Removes auth and roles fields from the user base type into a separate type. 
Then a whole bunch of cleaning things around here and there to fix build and test errors. 

## ❗ Note

Some things still in question like the role of the email field.
Cypress saves some headache by exposing how you could accidentally null out a user's sub.

## 🧪 Testing

1. Can build, login, and exercise actions
2. Tests pass

This is definitely something where you need to be really careful 😨 
Multiple reviews probably for the best

### CODE REVIEW

**Review by esizer — CHANGES_REQUESTED** [review]
This is great! Looks like you just need to look at some aliases in the `e2e` tests.

> **Note**
> Our profile pages don't need to re-render the router anymore 🎉

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -25,9 +25,17 @@ type LocalizedString {
   fr: String
 }
 
+type UserAuthInfo {
+  id: UUID!
+  email: Email
```
[the remark this answers is not in the document]
Comment (esizer) [reply api/graphql/schema.graphql:30]: We do assign it in the `AuthorizationContainer` context but I can't remember why 🤔 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/f24599f389a3bb19b2a05ea7a735bc53885fef6f/packages/auth/src/components/AuthorizationProvider.tsx#L28

@petertgiles Do you remember why we did this?

[the remark this answers is not in the document]
Comment (vd1992) [reply api/graphql/schema.graphql:30]: It is

image

[the remark this answers is not in the document]
Comment (tristan-orourke) [reply api/graphql/schema.graphql:30]: CreateAccountPage (and CreateAccountRedirect) was the only place I could find where we're actually using email from the Auth Context.

So, we could leave email here, but I just want to point out that we also have email in the BasicInfo object (to be renamed to ContactInfo?) and we've talked about wanting to minimize overlap.

[the remark this answers is not in the document]
Comment (vd1992) [reply api/graphql/schema.graphql:30]: Will need to refactor the account redirect to not rely on email if you want to keep it out of authInfo.

File: apps/e2e/package.json [hunk 2]
```diff
@@ -5,8 +5,8 @@
     "e2e": "npm run e2e:open",
     "e2e:open": "cypress open --e2e",
     "e2e:run": "npm run e2e:run:all",
-    "e2e:run:all": "cypress run --browser=electron --spec \"cypress/e2e/**/*${TEST_FILTER:-}*.cy.js\"",
-    "e2e:run:inspect": "cypress run --browser=electron --spec \"cypress/e2e/**/*${TEST_FILTER:-language-selection}*.cy.js\" --no-exit --headed"
+    "e2e:run:all": "cypress run --browser=electron --spec \"cypress/e2e/**/*${TEST_FILTER:-}*.cy.ts\"",
```
Comment (esizer) [inline apps/e2e/package.json:8]: Oops 😅

File: apps/web/src/pages/Users/UpdateUserPage/UpdateUserPage.tsx
Comment (vd1992) [inline apps/web/src/pages/Users/UpdateUserPage/UpdateUserPage.tsx:1]: Cypress helping out loads here.
The failing workflows were puzzling but eventually I got to the bottom of it and found something important to not have happen 😆 

Changing the shape of sub meant nothing appeared to load in for sub on this page, since it is typed as `Maybe<Sub>` it ending up null throws no issue. It isn't actually null, just present at `user.authInfo.sub` rather than `user.sub`

image

Then, when you go to submit, this function sees the empty field and submits a request to null out sub.
The field is empty because sub is being pulled from the wrong place.

image

This definitely feels a bit concerning, as you can inadvertently null out a user's sub while doing something as simple as updating a telephone number. If for whatever reason sub wasn't fetched (like in this case was moved but not update everywhere) or the form glitches. 
Feels like updating sub should be its separate thing so you don't accidentally do it. 

In other words, yeah @tristan-orourke, good idea with narrowed mutations and extracting important/special stuff out of type user.

File: apps/e2e/package.json (hunk 2 above)
Comment (vd1992) [reply apps/e2e/package.json:8]: Cleaning up crumbs 😛

File: apps/web/src/pages/Users/UpdateUserPage/UpdateUserPage.tsx
Comment (tristan-orourke) [reply apps/web/src/pages/Users/UpdateUserPage/UpdateUserPage.tsx:1]: Good catch!

File: api/graphql/schema.graphql (hunk 1 above)
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply api/graphql/schema.graphql:30]: You don't necessarily have to stop relying on email - you could also refactor this to get email from a different query. CreateAccountRedirect could run its own query, or we could make another hook (similar to useAuthorization) which holds some profile data from `me` query, like name and email and id.

[the remark this answers is not in the document]
Comment (vd1992) [reply api/graphql/schema.graphql:30]: Concerns from standup:
Second query for auth container/state just for email. 
Convenience versus best API practices

[the remark this answers is not in the document]
Comment (vd1992) [reply api/graphql/schema.graphql]: Okay, changes made!

File: apps/web/src/pages/Auth/CreateAccountPage/CreateAccountRedirect.tsx [hunk 3]
```diff
@@ -33,7 +35,7 @@ const CreateAccountRedirect = () => {
      *  - User has no email associated with account
      *  - User is not trying to go to the welcome page directly already
      */
-    if (loggedIn && authorizationLoaded && empty(email) && isToCreateAccount) {
+    if (loggedIn && !fetching && empty(email) && isToCreateAccount) {
```
Comment (tristan-orourke) [inline apps/web/src/pages/Auth/CreateAccountPage/CreateAccountRedirect.tsx:38]: I just realized that the `isLoaded` value from `useAuthorization` is actually `const isLoaded = !fetching && !stale`, so it may be a good idea to use both fetching and stale here as well.

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Just one small adjustment to how we check that we're ready to navigate

File: apps/web/src/pages/Auth/CreateAccountPage/CreateAccountRedirect.tsx (hunk 3 above)
Comment (vd1992) [reply apps/web/src/pages/Auth/CreateAccountPage/CreateAccountRedirect.tsx]: Sure, we can do that. 
I added in a reference to `stale` here in https://github.com/GCTC-NTGC/gc-digital-talent/pull/8297/commits/4dbe55f9f74fe94132b00e8fad114f08beea8f23

**Review by tristan-orourke — APPROVED** [review]

**Comment by vd1992** [comment]
@esizer 

You've still got changes requested, please re-review when you've got a chance or dismiss
